--- a/meson.build	2019-02-06 18:26:02.000000000 -0600
+++ b/meson.build	2019-02-09 23:17:54.808351343 -0600
@@ -31,7 +31,7 @@
 atk_req = '>= 2.5.3'
 
 # optional version requirements
-udev_req = '>= 228'
+udev_req = '>= 220'
 gudev_req = '>= 232'
 
 # wayland version requirements
@@ -178,7 +178,7 @@
   libgbm_dep = dependency('gbm', version: gbm_req)
   libinput_dep = dependency('libinput', version: libinput_req)
 
-  libsystemd_dep = dependency('libsystemd')
+  libsystemd_dep = dependency('libsystemd', required: false)
   if libsystemd_dep.found()
     logind_provider_dep = libsystemd_dep
   else
